tests/baselines/reference/checkJsxGenericTagHasCorrectInferences.errors.txt

```diff
@@ -17,6 +18,7 @@ tests/cases/conformance/jsx/file.tsx(13,54): error TS2322: Type '(a: { x: string
     let c = <GenericComponent initialValues={{ x: "y" }} nextValues={a => ({ x: a.x })} />; // No Error
     let d = <GenericComponent initialValues={{ x: "y" }} nextValues={a => a.x} />; // Error - `string` is not assignable to `{x: string}`
                                                          ~~~~~~~~~~
-!!! error TS2322: Type '(a: { x: string; }) => string' is not assignable to type '(cur: { x: string; }) => { x: string; }'.
-!!! error TS2322:   Type 'string' is not assignable to type '{ x: string; }'.
-!!! related TS6500 tests/cases/conformance/jsx/file.tsx:13:54: The expected type comes from property 'nextValues' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<GenericComponent<{ initialValues: { x: string; }; nextValues: {}; }, { x: string; }>> & { initialValues: { x: string; }; nextValues: {}; } & BaseProps<{ x: string; }> & { children?: ReactNode; }'
\ No newline at end of file
+!!! error TS2322: Type '(a: { x: string; }) => string' is not assignable to type '((a: { x: string; }) => string) & ((cur: { x: string; }) => { x: string; })'.

```

Comment:
This intersection is unfortunate. Is there a way to avoid adding `(a: { x: string }) => string` to the type?

Reply:
The target is `Props & BaseProps<Values>` where we've already infered `Props` as `{  initialValues: {x: string};
  nextValues: (cur: {x: string}) => string;
}` (since that's the entire attributes type as written, which is inferred to `Props`) and `BaseProps<Values>` as `interface BaseProps<T> {
  initialValues: T;
  nextValues: (cur: T) => T;
}` instantiated with `T = Values = {x: string}` (since that's what first matches `T` in the first context-free inference pass) - the intersection of both's `nextValues` member is the type we see here. Everything is as it should be.